cmd/controller/controller.go

```diff
@@ -124,15 +120,25 @@ func (c *CmdOpts) startController(ctx context.Context) error {
 	if err := dir.Init(c.K0sVars.CertRootDir, constant.CertRootDirMode); err != nil {
 		return err
 	}
+	// let's make sure run-dir exists
+	if err := dir.Init(c.K0sVars.RunDir, constant.RunDirMode); err != nil {
+		logrus.Fatalf("failed to initialize dir: %v", err)
```

Comment:
These could just return the (decorated?) error instead of relying on the exit functionality of `logrus.Fatalf`.